src/components/course-calendar/plan.cjsx

```diff
@@ -36,12 +40,10 @@ CoursePlan = React.createClass
     planId is @props.item.plan.id
 
   _isPlanNotMatchingRouteOpen: ->
-    {planId} = @context.router.getCurrentParams()
-    not @_doesPlanMatchesRoute() and @state.isViewingStats
+    not (@_doesPlanMatchesRoute()) and @refs.details?
 
   _isPlanMatchRouteNotOpen: ->
-    {planId} = @context.router.getCurrentParams()
-    @_doesPlanMatchesRoute() and not @state.isViewingStats
+    (@_doesPlanMatchesRoute()) and not @refs.details?
 
```

Comment:
These method names hurt my brain.  Which probably says more about my brain than the names, but still...

What I think they're doing is checking if the current route matches the given task, and if the modal is open.  If that's the case why not rename to `isMatchingRouteOpen`?  Or just `isCurrentStateValid()`, since it looks like they're used to determine if the stats should be updated or the modal closed below.